backend/capellacollab/projects/toolmodels/diagrams/routes.py

```diff
@@ -128,14 +128,20 @@ async def get_diagram(
     file_path = f"diagram_cache/{parse.quote(diagram_uuid, safe='')}.svg"
 
     try:
-        file_or_artifact = await handler.get_file_or_artifact(
+        res_job_id, date, data = await handler.get_file_or_artifact(
             trusted_file_path=file_path,
             logger=logger,
             job_name="update_capella_diagram_cache",
             job_id=job_id,
             file_revision=f"diagram-cache/{handler.revision}",
         )
-        return responses.SVGResponse(content=file_or_artifact[2])
+
+        if res_job_id == job_id:
```

Comment:
The `res_job_id` returned by the function is always the `job_id` when it's passed to the function, so the statement will always evaluate to true if a `job_id` is given. 

Instead, you can just check if a `job_id` was passed as parameter: 

```py
if job_id:
```